Communicate warning to users when no shortcodes with UI are registered

goldenapples opened the pull request:
This is a first attempt at addressing some of the issues pointed out in #711.

First off, if this plugin is activated, but no shortcodes are registered with Shortcake UI, an admin notice will be shown:

![screenshot from 2017-03-29 13-27-55](https://cloud.githubusercontent.com/assets/665992/24475183/ec156126-1483-11e7-94fa-de46820a9af5.png)

> The Shortcode UI plugin will not do anything unless UI is registered for shortcodes through a theme or plugins. For examples, see here.

with a link to [the wiki page here](https://github.com/wp-shortcake/shortcake/wiki/Shortcode-UI-Examples) with plugins which provide supported shortcodes.

Secondly, if there are no supported shortcodes available, the "Add Post Element" button will not be shown, as it doesn't do anything and is a source of potential confusion.

goldenapples committed:
Warn on activation if no supported shortcodes

Adds an admin notice on activation warning if no shortcodes with UI are available.

goldenapples committed:
Disable the media button if no shortcodes are registered

The "Add post element" button does nothing if no shortcode UIs are
registered. This prevents it from being attached to the editor if no
shortcodes are available to be inserted through it.

(Note that because `register_shortcode_ui` has access to the editor ID
as context, it's possible for shortcodes to be registered in some editor
contexts and not others. This handles that sort of... to the same degree
that `register_shortcode_ui` does, at least.)

goldenapples committed:
Fix some codesniff issues, better escaping

goldenapples committed:
Merge branch 'master' into 711-better-user-explanations

goldenapples committed:
Avoide fatals in PHP 5.2

Prevent throwing fatals in PHP 5.2 by casting to bool rather than
checking if $this->get_shortcodes is empty().

goldenapples committed:
Whitespace fix for code linting

szepeviktor reviewed (approved):
I confirm that it is a one-time notice and it is OK.

mattheu commented on inc/class-shortcode-ui.php:
```diff
@@ -81,6 +82,44 @@ private function setup_actions() {
+				'</p></div>' . "\n";
+		} else {
+			echo '<div class="notice notice-info is-dismissable"><p>' .
+				esc_html__( 'That\'s all! Try out the shortcode UI through the "Add Post element" button in the post edit screen.', 'shortcode-ui' ) .
```
I'm not so sure about 'Thats all'. I'm concerned it won't be immediately clear what its referring to. EG if the tab has been left open and the user comes back to it, the context of having just clicked activate is lost. 

How about `Shortcode UI installed. Try it out through the "Add Post element" button in the post edit screen."

'

goldenapples replied:
Good point. I'll update.

mattheu reviewed (commented):
Looks good. I commented with my thoughts on the notice text - but if you'd like to keep it as you have it - which is maybe a bit more casual, I'm OK with that.

goldenapples committed:
More clear, less informal, notice language

"That's all" is unclear in this context. Better to give some context as
to what the activation notice is actually referring to.
